crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -179,51 +183,52 @@ impl ModuleSubscriptionActor {
+    async fn broadcast_commit_event(&mut self, mut event: ModuleEvent) -> Result<(), DBError> {
         let auth = AuthCtx::new(self.owner_identity, event.caller_identity);
-
-        for subscription in &mut self.subscriptions {
-            let database_update = event.status.database_update().unwrap();
-            let incr = subscription
-                .queries
-                .eval_incr(&self.relational_db, tx, database_update, auth)?;
-
-            if incr.tables.is_empty() {
-                continue;
-            }
-
-            let message = TransactionUpdateMessage {
-                event: &mut event,
-                database_update: incr,
-            };
-            let mut message = CachedMessage::new(message);
-
-            for subscriber in subscription.subscribers() {
-                // rustc realllly doesn't like subscriber.send_message(message) here for weird
-                // lifetime reasons, even though it would be sound
-                let message = message.serialize(subscriber.protocol);
-                futures.push(subscriber.send(message).map(drop))
+        let relational_db = self.relational_db.clone();
+        let subscriptions = self.subscriptions.clone();
+        tokio::spawn(async move {
+            let mut tx = relational_db.begin_tx();
+            let subscriptions = subscriptions.read_arc();
+            let futures = FuturesUnordered::new();
+            for subscription in subscriptions.iter() {
+                let database_update = event.status.database_update().unwrap();
+                match subscription
+                    .queries
+                    .eval_incr(&relational_db, &mut tx, database_update, auth)
+                {
+                    Ok(incr) => {
+                        if incr.tables.is_empty() {
+                            continue;
+                        }
+                        let message = TransactionUpdateMessage {
+                            event: &mut event,
+                            database_update: incr,
+                        };
+                        let mut message = CachedMessage::new(message);
+
+                        for subscriber in subscription.subscribers() {
+                            // rustc realllly doesn't like subscriber.send_message(message) here for weird
+                            // lifetime reasons, even though it would be sound
+                            let message = message.serialize(subscriber.protocol);
+                            futures.push(subscriber.send(message).map(drop))
+                        }
+                    }
+                    Err(e) => log::error!("error occured in broadcasting commit event {:?}", e),
+                }
             }
-        }
-
-        futures.collect::<()>().await;
+            let ctx = ExecutionContext::incremental_update(relational_db.address());
+            relational_db.release_tx(&ctx, tx);
+            futures.collect::<()>().await;
```

Comment:
The order of the futures.collect() and the incremental_update call has been switched now.